Tweaks, Slashers balances for more players, Covenant

Xerkdot opened the pull request:
# --------GENERAL--------
- Increased max players to 20. (I recommend playing with 16 players, capped at 20 players to avoid possible unstability.)
- New lobby map file (more players spawns).
- "SlashCo.IsPositionLegalForSlashers" now only checks for near survivors (made this change to prevent impossible rounds for slashers on some maps and generators spots).
- Added SlashCo.LangTable for Covenant members abilities.
- Updated spanish lang file to match last update additions.
- Fixed confusing spanish translation for Offerings descriptions.

# --------OFFERINGS--------
- The Duality offering now requires 2 generators to escape.

# --------ITEMS--------
- Added two new items: Balkan Boost + Nightvision Goggles

# --------SLASHERS--------
--- **Abomignat:** 
- Increased Prowl speed (150 - 200) and Chase speed (293 - 325).
- Increased Chase duration (5s - 7s).
- Increased Lunge damage (50 - 100).
- Increased Crawling speed (350 - 400).

--- **Amogus:**
- Moved Gascan disguise position to the ground level.
- Fixed chase animation not being used.

--- **Bababooey:**
- Increased Chase speed (298 - 305).

--- **Borgmire:**
- Increased Chase speed (325 - 335).
- Decreased Chase cooldown (8s - 3s).
- Now he can freely move and aim when holding a survivor.

--- **Dolphinman:**
- Increased KillDelay (0.25s - 0.50s).
- Decreased hunt power gain on kill when hunting (25% - 15%).
- Decreased hunt power loss on standing (50% - 25%).
- Eyesight decreased. (3 - 2)
- Gains 25% hunt power when killing without hunting.
- KillDellay is increased to 5s when no hunting.
- Start gaining hunt power when the escape helicopter spawns.
- Added a breathing sound when hiding.

--- **Freesmiley:**
- Balkan Boost users are marked until the effect run out.
- Increased Chase duration (5s - 9s).

--- **Manspider:**
- Increased Prowl speed (150 - 250) and Chase speed (290 - 315).

--- **Sid:**
- Decreased KillDelay (7s - 5s)

--- **Thirsty:**
- Increased base Prowl speed (100 - 120) and base Chase speed (260 - 280).
- Decreased Pacified duration (same duration of Sid).
- Increased max milkies (4 - 6).
- After 4 milkies, Thirst meter will not increase anymore and will give you a permanent visibility buff.

--- **Trollge:**
- Can see the halos of Balkan Boost holders.
- On contact with a Balkan Boost, instantly change to stage 3.

--- **Tyler:**
- Beacon Arming activates destroyer mode.
- On spawn of the escape helicopter, goes perma destroyer.
- Improved tyler hide time duration.

--- **Watcher:**
- Reduced Prowl speed (200 - 185).

--- **Covenant:**
- Now spawns correctly all covenant members.
- Now all covenant members correctly enter chase mode when Covenant does.
- First kill summons "LTG Rocks", rest of victims will be summoned as "Cloaks".
- Properly added a minimum distance to be able to kill a player.
- "LTG Rocks" can now hit with the Saturn Stick (only on Chase).
- Added footstep sounds to covenant and his members.
- Reduced Chase duration (160s - 60s).
- Reduced Chase speed (297 - 275).
- Increased "Cloaks" Prowl speed (100 - 150).
- Fixed "Cloaks" ability, Tackle now stuns survivors for 1.2s, Cloak will remain freezed for 4.5s after using it.
- "Cloaks" can mark survivors when tackled.
- Changed Danger level to Devastating.

# --------PENDING TO FIX--------
- Find a replacement for the method i used to prevent players getting stuck (Cloaks Tackle) [The current method using "ply:SetImpervious()" will not be a problem if Mantibro/SlashCo/pull/27 is merged (at least the impervious fix)]
- ~~Show a lightning effect when hitted by Saturn Stick.~~
- ~~Make Cloaks "Tackle" ability work.~~

Xerkdot committed:
Pending changes exported from your codespace

textstack commented:
i will likely make changes to this and probably also wait until raphael is done working on their pr
also are *you* done with this pr considering the wip status of the covenant for you?

textstack commented:
actually, this is a pretty good opportunity to work on stuff and figure out git/github

it's a good idea to make this into a draft pull request if you are not done with it, it shows the intention to get the pr done while also signifying it is not ready to merge yet

getting the hang of git commits and pushing is also very convenient as it lets both of us do some cool stuff with your code progress and history

Xerkdot committed:
Update manspider.lua

Manspider can now Leap without chasing

Xerkdot commented:
> actually, this is a pretty good opportunity to work on stuff and figure out git/github
> 
> it's a good idea to make this into a draft pull request if you are not done with it, it shows the intention to get the pr done while also signifying it is not ready to merge yet
> 
> getting the hang of git commits and pushing is also very convenient as it lets both of us do some cool stuff with your code progress and history

alright thank you, i will work on finishing pending stuff and learn how to properly use github commits

textstack commented on gamemodes/slashco/gamemode/sh_shared.lua:
```diff
@@ -173,12 +173,6 @@ end
-			return false
-		end
-	end
-
```
The generator check for this was added because slashers like dolphinman and manspider can very easily fall into generator camping strategies for lower player counts. You might be able to check for generator count (it's a global int),  map size (slashco has a constant for it), or whether the generator is powered (they have nwbools for that) to figure out a way to get it to not do this when it'd be too much.

Xerkdot committed:
Manspider: fixed being able to leap while nested.

Xerkdot committed:
Borgmire buff: not freezed when holding a survivor

Xerkdot committed:
Rocks: added spark effect to target, increased hit damage + decreased hit distance

Xerkdot committed:
Amogus: fixed gascan disguise position (for real now) + added thirdperson

Xerkdot committed:
Covenant: Reduced chase duration and chase speed

Xerkdot committed:
Reduced chase duration and chase speed

Xerkdot committed:
Covenant Cloak: fixed tackle ability

Xerkdot committed:
Amogus: fixed chase animation not working.

Xerkdot committed:
Covenant: changed a few things

Cloaks now marks survivors, increased all chase speed, covenant can no longer cancel chase, increased hit distance for Rocks

Xerkdot committed:
Cloaks: increased a bit the ProwlSpeed

Xerkdot committed:
Cloaks: increased mark duration

Xerkdot committed:
Items: removed extras "AddCSLuaFile()"

Xerkdot committed:
Dolphinman: A few gameplay changes

- Eyesight decreased. (3 - 2)
- Gains 25% hunt power when killing without hunting.
- KillDellay is increased to 5s when no hunting.
- Start gaining hunt power when the escape helicopter spawns.
- Added a breathing sound when hiding.

Xerkdot committed:
Dolphinman: wrong file, whoopsies
